tests/testthat/test-return_linter.R

```diff
@@ -1521,3 +1471,231 @@ test_that("= assignments are handled correctly", {
     implicit_linter
   )
 })
+
+test_that("terminal switch() is handled correctly", {
+  implicit_linter <- return_linter()
+  implicit_msg <- rex::rex("Use implicit return behavior; explicit return() is not needed.")
+  explicit_linter <- return_linter(return_style = "explicit")
+  explicit_msg <- rex::rex("All functions must have an explicit return().")
+
+  no_return_lines <- trim_some("
+    foo <- function(x) {
+      switch(x,
+        a = 1,
+        b = 2
+      )
+    }
+  ")
+  expect_lint(no_return_lines, NULL, implicit_linter)
+  expect_lint(no_return_lines, list(explicit_msg, explicit_msg), explicit_linter)
+
+  outer_return_lines <- trim_some("
+    foo <- function(x) {
+      return(switch(x,
+        a = 1,
+        b = 2
+      ))
+    }
+  ")
+  expect_lint(outer_return_lines, implicit_msg, implicit_linter)
+  expect_lint(outer_return_lines, NULL, explicit_linter)
+
+  partial_return_lines <- trim_some("
+    foo <- function(x) {
+      switch(x,
+        a = return(1),
+        b = 2
+      )
+    }
+  ")
+  expect_lint(partial_return_lines, implicit_msg, implicit_linter)
+  expect_lint(partial_return_lines, explicit_msg, explicit_linter)
+
+  all_return_lines <- trim_some("
+    foo <- function(x) {
+      switch(x,
+        a = return(1),
+        b = return(2)
+      )
+    }
+  ")
+  expect_lint(all_return_lines, list(implicit_msg, implicit_msg), implicit_linter)
+  expect_lint(all_return_lines, NULL, explicit_linter)
```

Comment:
Should the behavior of `explicit_linter` here depend on `allow_implicit_else`, i.e. force a `return()` in the default case if the latter is `FALSE`?

Reply:
interesting Q... I lean towards yes. @MEO265 any thoughts?

Reply:
Ditto actually for implicit return style requiring an unnamed arg to `switch()`.

Reply:
OK, `allow_implicit_else` also covers whether the default case is required in a terminal `switch()` statement.